src/main/scala/com/yotpo/metorikku/test/TestUtil.scala

```diff
@@ -0,0 +1,114 @@
+    var res = List[mutable.LinkedHashMap[String, Any]]()
+    val indexesToCollect = errorsIndexArr.length match {
+      case 0 => sortedExpectedRows.indices
+      case _ =>
+        errorsIndexArr.contains(sortedExpectedRows.length - 1) match {
+          case true => errorsIndexArr
+          case _ => errorsIndexArr
+        }
+    }
+    for (index <- indexesToCollect) {
+      var tempRes = mutable.LinkedHashMap[String, Any]()
+      val resIndx = index + 1
+      tempRes += ("row_number" -> resIndx)
+      val sortedRow = sortedExpectedRows(index)
+      for (col <- sortedRow.keys) {
+        tempRes += (col -> sortedRow(col))
+      }
+      res = res :+ tempRes
+    }
+    res
+  }
+
+  def getMapFromDf(dfRows: DataFrame): List[Map[String, Any]] = {
+    dfRows.rdd.map {
+      dfRow =>
+        dfRow.getValuesMap[Any](dfRow.schema.fieldNames)
+    }.collect().toList
+  }
+
+
+  def getLongestValueLengthPerKey(results: List[Map[String, Any]]): Map[String, Int] = {
+    // the  keys of head result should be from the expected format
+    // (actual results might have fields that are missing in the expected results (those fields need to be ignored)
+    results.head.keys.map(colName => {
+      val resColMaxLength = results.maxBy(c => {
+        if (c(colName) == null) {
+          0
+        } else {
+          c(colName).toString.length
+        }
+      })
+      colName -> resColMaxLength.get(colName).toString.length
+    }
+    ).toMap
+  }
+
+  def addLongestWhitespaceRow(mapList: List[RowObject],
+                              longestRowMap: Map[String, Int]): List[RowObject] = {
+    val longestRow = longestRowMap.map { case (col, maxColValLength) =>
+      val sb = new StringBuilder
+      for (_ <- 0 to maxColValLength) {
+        sb.append(" ")
+      }
+      col -> sb.toString
+    }
+    mapList :+ RowObject(longestRow, mapList.size + 1)
+  }
+
+  def getMismatchedVals(expectedResultRow: Map[String, Any], actualResultRow: Map[String, Any],
+                        mismatchingCols: ArrayBuffer[String]): ArrayBuffer[String] = {
+    var res = ArrayBuffer[String]()
+    for (mismatchCol <- mismatchingCols) {
+      res +:= s"${mismatchCol} - Expected = ${expectedResultRow(mismatchCol)}, Actual = ${actualResultRow(mismatchCol)}"
+    }
+    res
+  }
+
+  def getMismatchingColumns(actualRow: Map[String, Any], expectedRowCandidate: Map[String, Any]): ArrayBuffer[String] = {
+    var mismatchingCols = ArrayBuffer[String]()
+    for (key <- expectedRowCandidate.keys) {
+      val expectedValue = Option(expectedRowCandidate.get(key))
+      val actualValue = Option(actualRow.get(key))
+      // TODO: support nested Objects and Arrays
+      if (expectedValue.toString != actualValue.toString) {
+        mismatchingCols += key
+      }
+    }
+    mismatchingCols
+  }
+
+  def getDfShowStr(df: DataFrame, size: Int, truncate: Boolean, lastRowIndexStr: String): String = {
```

Comment:
lastRowIndexStr.... I don't like that part... maybe add an option to print only the first X lines and then the logic could be where you added the weird extra line
it shouldn't be in the generic print function